pkg/queue/sharedmain/main.go

```diff
@@ -169,6 +170,8 @@ func Main(opts ...Option) error {
 	d := Defaults{
 		Ctx: signals.NewContext(),
 	}
+	pendingRequests := atomic.Int32{}
+	pendingRequests.Store(0)
```

Comment:
See above comment - let's merge this into a special http.Handler struct counting requests - that is created in the `mainHandler` function